frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/update-utils.ts

```diff
@@ -439,21 +539,74 @@ export const applyChange = (
   op: UpdateOperation,
 ): CleanupResults => {
   const { type, data } = op;
-  const { tasks, listTasks } = taskGroup;
+  const { tasks, listTasks, finallyTasks, finallyListTasks } = taskGroup;
 
   switch (type) {
     case UpdateOperationType.ADD_LIST_TASK:
-      return addListNode(tasks, listTasks, data as UpdateOperationAddData);
+      return addListNode(
```

Comment:
Maybe we also just pass in `taskGroup` at this point. I know it has the selectedIds but that's immaterial I think. Having two data to each method makes it easier than managing multiple separate lists that all came from the same object; it's getting a bit unruly. 

ie `return methodName(taskGroup, data as Whatever)`

Reply:
Ah nice, it makes sense. updated